kai/service/llm_interfacing/model_provider.py

```diff
@@ -51,9 +52,16 @@ def __init__(self, config: KaiConfigModels):
 
             case "ChatIBMGenAI":
                 model_class = LangChainChatInterface
-
+                if os.getenv("DEMO_MODE", "false").lower() == "true":
+                    api_key = os.getenv("GENAI_KEY", "dummy_value")
+                    api_endpoint = os.getenv("GENAI_API", "dummy_endpoint")
```

Comment:
I think we need to remove this redefinition of the API endpoint.  

As I'm testing today I'm seeing behavior for the case where we have valid credentials is not quite working in Demo Mode as we typically would not set the GENAI_API variable.  

I think to move fwd all we want to do is set the GENAI_KEY.

I will try to confirm later today/tomorrow.

@Myrausman apologies for the extensive back and forth on reviews with this PR.  The change you have is good, there have just been a few subtle issues exposed with testing which I assume you aren't able to do as the credentials for IBM BAM service is limited.